rfcs/0016-fuzz-attributes.md

```diff
@@ -0,0 +1,192 @@
+- Feature Name: `fuzz_attributes`
+- Start Date: 2025-09-09
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
+- Sway Issue: [FueLabs/sway#0000](https://github.com/FuelLabs/sway/issues/001)
+
+# Summary
+
+[summary]: #summary
+
+This RFC introduces parsing support for `#[fuzz]` and `#[fuzz_param]` attributes in Sway to enable fuzz testing capabilities. These attributes allow developers to define fuzz test functions and specify parameters for fuzz testing operations.
+
+# Motivation
+
+[motivation]: #motivation
+
+Fuzz testing is a critical software testing technique that helps discover edge cases, security vulnerabilities, and unexpected behaviors by feeding random or semi-random inputs to functions. Currently, Sway lacks native support for fuzz testing attributes, making it difficult for developers to implement comprehensive testing strategies that include fuzzing.
+
+The introduction of `#[fuzz]` and `#[fuzz_param]` attributes will:
+- Enable systematic fuzz testing of Sway functions
+- Provide a standardized way to define fuzz test parameters
+- Improve code quality and security through automated testing of edge cases
+- Align with modern testing practices found in other programming languages
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+## Basic Fuzz Testing
+
+The `#[fuzz]` attribute marks a function as a fuzz test. This function will be executed with randomly generated inputs to test for unexpected behaviors:
+
+```sway
+#[fuzz]
+fn fuzz_my_function(input: u64) {
+    // Test logic that will be executed with random u64 values
+    let result = my_function(input);
+    // Add assertions or checks here
+}
+```
+
+## Parameterized Fuzz Testing
+
+The `#[fuzz_param]` attribute allows you to specify parameters for more controlled fuzz testing:
+
+```sway
+#[fuzz]
+#[fuzz_param(name = "input", iteration = 1000)]
+#[fuzz_param(name = "seed", min_val = 0, max_val = 100)]
+fn fuzz_with_params(input: u64, seed: u32) {
+    // This will run 1000 iterations with:
+    // - input: random u64 values
+    // - seed: random u32 values between 0 and 100
+}
+```
+
+## Usage Constraints
+
+- A function cannot have both `#[test]` and `#[fuzz]` attributes
+- The `#[fuzz]` attribute expects no arguments
+- Multiple `#[fuzz_param]` attributes are allowed per function
+- Only one `#[fuzz]` attribute is allowed per function
+- `#[fuzz_param]` supports the following parameters:
+  - `name`: The parameter name to configure
+  - `iteration`: Number of test iterations
```

Comment:
Just thinking would it make more sense to define this on the `#[fuzz]` attribute? 

so 

```sway
#[fuzz, iteration = 1000]
```